14.0 [MIG] account_invoice_pricelist: Migration to 14.0

hildickethan opened the pull request:
Standard migration to 14.0

Applied [this change](https://github.com/OCA/account-invoicing/commit/defd56407d9c719aab4923ea1ba18df19f54174c#diff-f63da6ff419afd7f33e73cf400ec73fe25d83624e5e9ab30c92e4d88f7a4a47c) from 12.0 as well

legalsylvain committed:
new module account_invoice_pricelist;

oca-transbot committed:
OCA Transbot updated translations from Transifex

hbrunn committed:
use pricelist when calculating prices

sergio-teruel committed:
account_invoice_pricelist: Migration to v9

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

rvalyi committed:
account_invoice_pricelist: Migration to 10.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

amcor committed:
account_invoice_pricelist: Migration to 11.0

oca-travis committed:
Update account_invoice_pricelist.pot

OCA-git-bot committed:
README.rst

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (5 of 5 strings)

Translation: account-invoicing-11.0/account-invoicing-11.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-11-0/account-invoicing-11-0-account_invoice_pricelist/de/

OCA-git-bot committed:
README.rst

Nikul-Chaudhary committed:
[MIG] account_invoice_pricelist v11 to v12

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update account_invoice_pricelist.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_pricelist/

OCA-git-bot committed:
[ADD] icon.png

bealdav committed:
FIX acc_invoice_pricelist: tax include in multicompany

OCA-git-bot committed:
account_invoice_pricelist 12.0.1.0.1

bealdav committed:
FIX acc_invoice_pricelist: recompute tax after update pricelist

OCA-git-bot committed:
account_invoice_pricelist 12.0.1.0.2

am-technix committed:
Translated using Weblate (French)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_pricelist/fr/

OCA-git-bot committed:
[UPD] README.rst

legalsylvain committed:
[FIX] bad display on invoices form

OCA-git-bot committed:
account_invoice_pricelist 12.0.1.0.3

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (5 of 5 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_pricelist/pt_BR/

Commit:
Added translation using Weblate (Spanish (Chile))

Commit:
Translated using Weblate (Spanish (Chile))

Currently translated at 100.0% (5 of 5 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_pricelist/es_CL/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (5 of 5 strings)

Translation: account-invoicing-12.0/account-invoicing-12.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-12-0/account-invoicing-12-0-account_invoice_pricelist/hr/

manuelregidor committed:
[MIG] account_invoice_pricelist: black, isort, prettier

manuelregidor committed:
[MIG]account_invoice_pricelist: Migration to 13.0

manuelregidor committed:
[13.0][IMP] Supports multicurrency and different types of discounts

oca-travis committed:
[UPD] Update account_invoice_pricelist.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_pricelist/

Commit:
[IMP] account_invoice_pricelist: black, isort, prettier

Commit:
[14.0][MIG] account_invoice_pricelist: Migration to 14.0

hildickethan commented:
I don't know what the test errors are caused by, from what I could tell the code works fine but I fail to see where the expected values come from. My guess is that the USD to EUR rates aren't what they were previously in 13.0 or whenever the module was migrated, I'm not sure how Odoo gets them

If anyone could shed some light on the tests it would be appreciated

kittiu committed:
[13.0][FIX] account_invoice_pricelist, fix test currency rate

kittiu commented:
You can apply this patch https://github.com/OCA/account-invoicing/pull/875

kittiu reviewed (approved):
LGTM

kittiu commented:
There are some change in the PR https://github.com/OCA/account-invoicing/pull/875
Can you take a look please.

OCA-git-bot committed:
account_invoice_pricelist 13.0.1.0.1

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (6 of 6 strings)

Translation: account-invoicing-13.0/account-invoicing-13.0-account_invoice_pricelist
Translate-URL: https://translation.odoo-community.org/projects/account-invoicing-13-0/account-invoicing-13-0-account_invoice_pricelist/es/

hildickethan commented:
@kittiu since your PR is now merged, I re-applied the history of commits so everything should be ok now

Zar21 committed:
[14.0][FIX] account_invoice_pricelist: Send invoice_date with context

The invoice date was not being sent to the function that looks for the tax that has to be applied and therefore it was looking for the current date

gdgellatly commented on account_invoice_pricelist/models/account_move.py:
```diff
@@ -0,0 +1,228 @@
+            ):
+                price, rule_id = pricelist_item.base_pricelist_id.with_context(
+                    uom=uom.id
+                ).get_product_price_rule(product, qty, self.order_id.partner_id)
```
where/what is this order_id field?

This seems basically a cut and paste of the sale order line code. Anyway it should probably be move_id.

gdgellatly reviewed (commented):
The thing I don't like about modules like this, is they cut and paste so much, and short of using a new sale/line object and setting up and calling the functions it is a bit unavoidable.

But then my preference is that in the tests that where the code is copied, you take a hash of the copied function and if that hash changes you get a failure in tests indicating that the function you copied / largely copied has received a fix. 

e.g.
```python
import hashlib
import inspect

from odoo.tests.common import TransactionCase

from odoo.addons.product.models.product_pricelist import Pricelist as upstream

# mostly compatible, only bug fixes different, except last entry which is base
# other hashes may be valid but not tested, more of an early warning system
# for changed behaviour

VALID_HASHES = ["3b8538c9c617daf752b5ce8d588468bb"]


class TestProductPricelistHash(TransactionCase):
    def test_upstream_file_hash(self):
        """Test that copied upstream function hasn't received fixes"""
        func = inspect.getsource(upstream._compute_price_rule).encode()
        func_hash = hashlib.md5(func).hexdigest()
        self.assertIn(func_hash, VALID_HASHES)
```

hildickethan commented:
I wil lfinish updating the hash check on monday, I need the order_id change for now

hildickethan commented:
@gdgellatly Sorry for the late response and changes, I implemented your hash check for the primarily copied function as well as changed that order_id field that has been wrong since that function was copied in 13.0

dreispt commented:
/ocabot merge nobump
